Source/WebGPU/WebGPU/PresentationContext.mm

```diff
@@ -36,6 +36,7 @@
 Ref<PresentationContext> Device::createSwapChain(PresentationContext& presentationContext, const WGPUSwapChainDescriptor& descriptor)
 {
     presentationContext.configure(*this, descriptor);
+    presentationContext.ref();
```

Comment:
Why is this necessary? The input to this function is a PresentationContext& and the output is a Ref<PresentationContext>. So the next line return statement will invoke the constructor of Ref that will increment the ref count. So I think this change will actually bump the ref count by 2, not 1.

Also, suspicious that every ref needs to be matched with a deref but this one doesn't seem to be.

Reply:
Good observation, I will take a closer look.

Reply:
Oh I think I found the root cause now. The destructor and destroy were both calling release. `wgpuSurfaceRelease` doesn't actually release the backing, it just decrements the ref count. So after calling this I now set the backing to null so it can't be released again.